### PR TITLE
 update gerrit auth provider to add credentials for gerrit event-streams

### PR DESCRIPTION
**Pull request by ankul-shippable**
https://github.com/Shippable/admiral/issues/2629

### CODE REVIEW

File: api/systemIntegrations/post.js [hunk 1]
```diff
@@ -274,6 +271,14 @@ function _postProvider(bag, next) {
     name = name.replace('Keys', '');
   else if (name.endsWith('BasicAuth'))
     name = name.replace('BasicAuth', '');
+  bag.providerName = name;
+
+  if (!bag.reqBody.data.url) {
```
Comment (rageshkrishna) [inline api/systemIntegrations/post.js]: Shouldn't these checks happen in checkInputParams?

File: api/systemIntegrations/put.js [hunk 2]
```diff
@@ -246,6 +243,14 @@ function _postProvider(bag, next) {
     name = name.replace('Keys', '');
   else if (name.endsWith('BasicAuth'))
     name = name.replace('BasicAuth', '');
+  bag.providerName = name;
+
+  if (!bag.reqBody.data.url) {
```
Comment (rageshkrishna) [inline api/systemIntegrations/put.js]: Shouldn't these checks happen in checkInputParams?

File: api/systemIntegrations/put.js [hunk 3]
```diff
@@ -265,6 +270,35 @@ function _postProvider(bag, next) {
   );
 }
 
+function _genSshKeys(bag, next) {
```
Comment (rageshkrishna) [inline api/systemIntegrations/put.js]: Why is this necessary during the PUT? If the keys were not generated, the integration would not have been created to begin with, right?

File: api/systemIntegrations/post.js (hunk 1 above)
Comment (ankul-shippable) [reply api/systemIntegrations/post.js]: this is added here because initially we create the sysInt without any url. so if the url is not present then we dont create the provider. If it is present, then we create the provider.

File: api/systemIntegrations/put.js (hunk 2 above)
Comment (ankul-shippable) [reply api/systemIntegrations/put.js]: this is added here because initially we create the sysInt without any url. so if the url is not present then we dont create the provider. If it is present, then we create the provider.

File: api/systemIntegrations/put.js (hunk 3 above)
Comment (ankul-shippable) [reply api/systemIntegrations/put.js]: ok, removing it.

File: api/systemIntegrations/post.js (hunk 1 above)
Comment (rageshkrishna) [reply api/systemIntegrations/post.js]: Ok, makes sense. I misread the diff.
